README.md

```diff
@@ -189,54 +189,71 @@ using the installation ID of the GitHub app and authenticate with the OAuth meth
-use `UnauthenticatedRateLimitedTransport`.
-
-The returned `Response.Rate` value contains the rate limit information
+GitHub imposes rate limits on all API clients. The [primary rate limit](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api#about-primary-rate-limits)
+is the limit to the number of REST API requests that a client can make within a
+specific amount of time. This limit helps prevent abuse and denial-of-service
+attacks, and ensures that the API remains available for all users. Some
+endpoints, like the search endpoints, have more restrictive limits.
+Unauthenticated clients may request public data but have a low rate limit,
+while authenticated clients have rate limits based on the client
+identity.
+
+In addition to primary rate limits, GitHub enforces [secondary rate limits](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api#about-secondary-rate-limits)
+in order to prevent abuse and keep the API available for all users.
+Secondary rate limits generally limit the number of concurrent requests that a
+client can make.
+
+The client returned `Response.Rate` value contains the rate limit information
 from the most recent API call. If a recent enough response isn't
-available, you can use `RateLimits` to fetch the most up-to-date rate
-limit data for the client.
+available, you can use the client `RateLimits` service to fetch the most
+up-to-date rate limit data for the client.
 
-To detect an API rate limit error, you can check if its type is `*github.RateLimitError`:
+To detect a primary API rate limit error, you can check if the error is a
+`RateLimitError`.
 
 ```go
 repos, _, err := client.Repositories.List(ctx, "", nil)
-if _, ok := err.(*github.RateLimitError); ok {
-	log.Println("hit rate limit")
+var rateErr *github.RateLimitError
+if errors.As(err, &rateError) {
+  log.Printf("hit primary rate limit, used %d of %d\n", rateErr.Rate.Used, rateErr.rate.Limit)
 }
 ```
 
-Learn more about GitHub rate limiting in
-["REST API endpoints for rate limits"](https://docs.github.com/en/rest/rate-limit).
-
-In addition to these rate limits, GitHub imposes a secondary rate limit on all API clients.
-This rate limit prevents clients from making too many concurrent requests.
-
-To detect an API secondary rate limit error, you can check if its type is `*github.AbuseRateLimitError`:
+To detect an API secondary rate limit error, you can check if the error is an
+`AbuseRateLimitError`.
 
 ```go
 repos, _, err := client.Repositories.List(ctx, "", nil)
-if _, ok := err.(*github.AbuseRateLimitError); ok {
-	log.Println("hit secondary rate limit")
+var rateErr *github.AbuseRateLimitError
+if errors.As(err, &rateError) {
+  log.Printf("hit secondary rate limit, retry after %v\n", rateErr.RetryAfter)
 }
 ```
 
-Alternatively, you can block until the rate limit is reset by using the `context.WithValue` method:
+If you hit the primary rate limit, you can use the `SleepUntilPrimaryRateLimitReset`
```

Comment:
I didn't find any `SleepUntilPrimaryRateLimitReset` method. Is it a mistake?

Maybe you mean `github.SleepUntilPrimaryRateLimitResetWhenRateLimited`?

Reply:
> I didn't find any SleepUntilPrimaryRateLimitReset method. Is it a mistake?

It's not a method, the code below shows what it is and how to use it.

> Maybe you mean github.SleepUntilPrimaryRateLimitResetWhenRateLimited?

Everything in this module has a `github` prefix using the default import so I think this would be unnecessary.

Reply:
What I mean, is `SleepUntilPrimaryRateLimitReset` != `SleepUntilPrimaryRateLimitResetWhenRateLimited`